### PR TITLE
Update all dependencies to their latest versions

### PR DESCRIPTION
**Pull request by paazmaya**
According to https://david-dm.org/nodejs/nodejs.org there is also security issue with `marked` which has not been fixed.
However all dependencies are updated to their latest versions via this PR

### CODE REVIEW

**Comment by fhemberger** [comment]
Have you checked for any breaking changes, especially in `semver-major` updates?

File: package.json [hunk 1]
```diff
@@ -31,42 +31,42 @@
...
     "changelog-url": "1.0.2",
-    "cheerio": "0.19.0",
-    "chokidar": "1.2.0",
-    "handlebars": "4.0.4",
-    "html-to-text": "^1.5.0",
-    "js-yaml": "^3.4.5",
+    "cheerio": "0.20.0",
+    "chokidar": "1.4.3",
+    "handlebars": "4.0.5",
+    "html-to-text": "^2.1.0",
```
Comment (lpinca) [inline package.json:39]: Is this safe? What is the breaking change here?

File: package.json [hunk 2]
```diff
@@ -31,42 +31,42 @@
...
+    "metalsmith-layouts": "1.6.4",
     "metalsmith-markdown": "0.2.1",
-    "metalsmith-metadata": "0.0.2",
-    "metalsmith-permalinks": "0.4.0",
-    "metalsmith-prism": "2.1.1",
-    "metalsmith-stylus": "1.0.0",
+    "metalsmith-metadata": "0.0.4",
+    "metalsmith-permalinks": "0.5.0",
+    "metalsmith-prism": "2.2.0",
+    "metalsmith-stylus": "2.0.0",
```
Comment (lpinca) [inline package.json:52]: What are the breaking changes?

File: package.json [hunk 3]
```diff
@@ -31,42 +31,42 @@
...
+    "st": "1.1.0",
     "strftime": "0.9.2"
   },
   "devDependencies": {
     "faucet": "^0.0.1",
-    "nock": "^2.17.0",
+    "nock": "^8.0.0",
     "pre-commit": "^1.1.2",
-    "proxyquire": "^1.7.3",
+    "proxyquire": "^1.7.4",
```
Comment (lpinca) [inline package.json:68]: Unnecessary change.

File: package.json [hunk 4]
```diff
@@ -31,42 +31,42 @@
...
     "standard": "^6.0.8",
-    "tape": "^4.2.2"
+    "tape": "^4.5.1"
```
Comment (lpinca) [inline package.json:70]: Ditto.

File: package.json [hunk 5]
```diff
@@ -31,42 +31,42 @@
...
     "metalsmith-yearly-pagination": "2.0.0",
     "ncp": "2.0.0",
-    "node-geocoder": "^3.4.1",
-    "node-version-data": "1.0.0",
-    "octonode": "0.7.4",
-    "request": "^2.67.0",
+    "node-geocoder": "^3.9.1",
+    "node-version-data": "1.0.1",
+    "octonode": "0.7.5",
+    "request": "^2.72.0",
```
Comment (lpinca) [inline package.json:58]: Not needed, caret versioning covers this.

File: package.json [hunk 6]
```diff
@@ -31,42 +31,42 @@
...
+    "metalsmith-permalinks": "0.5.0",
+    "metalsmith-prism": "2.2.0",
+    "metalsmith-stylus": "2.0.0",
     "metalsmith-yearly-pagination": "2.0.0",
     "ncp": "2.0.0",
-    "node-geocoder": "^3.4.1",
-    "node-version-data": "1.0.0",
-    "octonode": "0.7.4",
-    "request": "^2.67.0",
+    "node-geocoder": "^3.9.1",
```
Comment (lpinca) [inline package.json:55]: Unnecessay change.

**Comment by lpinca** [comment]
This is good, I also wanted to do this, but it's risky, we need to be sure that we are not breaking things.

File: package.json [hunk 7]
```diff
@@ -31,42 +31,42 @@
...
-    "cheerio": "0.19.0",
-    "chokidar": "1.2.0",
-    "handlebars": "4.0.4",
-    "html-to-text": "^1.5.0",
-    "js-yaml": "^3.4.5",
+    "cheerio": "0.20.0",
+    "chokidar": "1.4.3",
+    "handlebars": "4.0.5",
+    "html-to-text": "^2.1.0",
+    "js-yaml": "^3.6.0",
```
Comment (lpinca) [inline package.json:40]: Same thing, not necessary.

File: package.json (hunk 1 above)
Comment (stevemao) [inline package.json:39]: We should pin this too

File: package.json (hunk 7 above)
Comment (stevemao) [inline package.json:40]: we should pin all deps and rely on future greenkeeper.io

Comment (lpinca) [inline package.json:40]: @stevemao https://github.com/nodejs/nodejs.org/issues/377

**Comment by stevemao** [comment]
I suggest we only bump minor versions here and leave major version bump in other PRs.

File: package.json [hunk 8]
```diff
@@ -31,42 +31,42 @@
...
   "dependencies": {
-    "autoprefixer-stylus": "0.8.1",
+    "autoprefixer-stylus": "0.9.2",
     "changelog-url": "1.0.2",
-    "cheerio": "0.19.0",
-    "chokidar": "1.2.0",
-    "handlebars": "4.0.4",
-    "html-to-text": "^1.5.0",
-    "js-yaml": "^3.4.5",
+    "cheerio": "0.20.0",
```
Comment (stevemao) [inline package.json:36]: Note for v0, a minor version bump contains breaking changes.

File: package.json (hunk 7 above)
Comment (stevemao) [inline package.json:40]: I know, that's why I said "future" :)

Comment (stevemao) [inline package.json:40]: If we aren't going with greenkeeper.io, I suggest to use `~` for all deps.

**Comment by lpinca** [comment]
I looked at the semver major updates and everything seems safe:
- `cheerio@0.20.0` → https://github.com/cheeriojs/cheerio/blob/master/History.md#0200--2016-02-01
- `html-to-text@2.1.0` → https://github.com/werk85/node-html-to-text/blob/master/CHANGELOG.md#version-200
- `metalsmith-stylus@2.0.0` → https://github.com/esundahl/metalsmith-stylus/pull/15

If there are no objections I would merge this and file a new pr where all deps (except dev dependencies) are pinned down or use tilde versioning.

**Comment by ghost** [comment]
yeah, looks safe to me as well. no breaking changes that affect our usage

**Comment by paazmaya** [comment]
I did not check anything other than running `npm test` and checking that site got served as it was before.
